### ISSUE TITLE
documentation on size_t typical value appears incorrect

### ISSUE THREAD

**Issue by jetmonk** [cffi/cffi#387]
In the callback tutorial in cffi-manual.texinfo it says:

> _size_t is almost always an unsigned int._  You can get this
and many other types using feature tests for your system by using
cffi-grovel.

I believe "almost always" isn't really correct.   I believe that size_t is  more typically the same size as a pointer, but even this isn't always true.   Exceptions  to sizeof(size_t)=sizeof(pointer)  I've found in searching are MS-DOS, IBM  AS/400, but I _think_  it holds for any modern Unix.

The assumption that size_t is an unsigned-int failed on arm64 Darwin, where an unsigned-int is 32 bits, but a pointer (and size_t) is 64 bits.   From the source below, it also seems incorrect for some Linux and Windows flavors.  

This document bug isn't a bug in cffi, because cffi does not define size_t itself.  

To avoid groveling, I've taken to defining size_t as a :uint32 or :uint64 based on the foreign-type-size of a :pointer using a compiler macro, which I hope is correct for modern Linux, Darwin, and Windows.   

One source for types on a some systems: http://synfare.com/JO59hw/hwdocs/sizes.html

**Comment by luismbo** [comment]
Yes, that statement didn't age very well. 32-bit architectures were very common back then, IIRC. Anyway, CFFI now provides a `:size` type. We guess using a similar heuristic to yours and the test suite checks that it's correct.

https://github.com/cffi/cffi/blob/5bfca29deb8b4c214a86ccf37279cc5cea2151e1/src/types.lisp#L1068
https://github.com/cffi/cffi/blob/5bfca29deb8b4c214a86ccf37279cc5cea2151e1/tests/misc-types.lisp#L309

**Comment by sionescu** [comment]
We'd really need to use comething like autoconf and its `config.guess` script, to fetch the system build triple.

**Comment by jetmonk** [comment]
Thank you;  :size was exactly what I needed.  I did apropos size-t / sizet / size_t so I didn't find it.   

Given how common the need for size_t is, it might be useful to add :size to Sec 3.1 of the manual (Built-in types) with a note that this is size_t (to make size_t appear in searches).

**Comment by luismbo** [comment]
:size is listed in the manual, but missing a reference to size_t. That's a good suggestion. Would you like to send a pull request?

**Comment by jetmonk** [comment]
To my chagrin, I was relying on outdated online quicklisp documentation.       Anway, I submitted pull request #388 with changes for both manuals.
